prysm/io.py

```diff
@@ -1363,3 +1363,96 @@ def read_sigfit_rigidbody(file):
             'dR': dR
         }
     return out
+
+def read_codev_int(filename):
+    """Read the content of a CodeV INT (.int/.cvg) file.
+    
+    Parameters
+    ----------
+    file : path_like
+        path to a file
+    
+    Returns
+    -------
+    'dict'
+        dictionary with keys: phase, intensity, meta
+
+    """
+    with open(filename, 'rt') as f:
+        # skip headerlines starting with "!"
+        while True:
+            title = f.readline()
+            if not title.lstrip().startswith('!'):
+                break
+        # read parameter line
+        param = f.readline().split()
+        # parse parameters
+        if param[0] == 'GRD':
+            num_cols = int(param[1])
+            if param[2] == 'R':
+                #radial grid data
+                #not implemented
+                raise ValueError('Radial Grid Data INT-File type not supported.')
+            else:
+                num_rows = int(param[2])
+            datatype = param[3]    #SUR | WFR | FIL
+            #need to test now because different orders are allow
+            p=4
+            while p < len(param):
+                if param[p].upper() == 'WVL':
+                    wavelength = float(param[p+1])
+                if param[p].upper() == 'NMB':
+                    pass
+                if param[p].upper() == 'SSZ':
+                    scale_size = float(param[p+1])
+                if param[p].upper() == 'NDA':
+                    nda = int(param[p+1])
+                if param[p].upper() == 'XSC':
+                    x_scale = float(param[p+1])
+                p += 1    
+            #read grid data
+            grdlist=[]
+            while True:
+                line = f.readline()
+                if not line:
+                    break
+                int_line = [int(i) for i in line.split()]
+                grdlist.extend(int_line)
+            if not len(grdlist) == num_cols*num_rows:
+                raise ValueError('Error during INT-file import. Length of list does not match row*cols')
+            if (datatype == 'WFR') or (datatype == 'SUR') :
+                # return values in nm (zygo convention, fit to prysm)
+                phase = e.array(grdlist).astype(config.precision)
+                phase[phase == nda] = e.nan
+                phase = phase*wavelength*1e3/scale_size
+                phase = phase.reshape((num_rows, num_cols))
+                phase = e.flipud(phase)
+            else:
+                #for "FIL"
+                raise ValueError('Intensity apodization filter data (FIL) INT-File type not supported')
+        elif (param[0] == 'ZRN') or (param[0] == 'ZRF'):
```

Comment:
could you replace all of these as one else clause, and f-string interpolate `param[0]` into the ValueError?

Reply:
An INT-File has several options and I would like to keep the "not supported" else clauses seperate if me (or someone else) would like to extend the import modul in the future. Especially the ZRN (Fringe Zernike) or ZFR (Standard Zernike) could be easily implemented as this format contains the Zernike Coefficients and prysm could generate the Interferogram out of that information.